### PR TITLE
Try failing fast if prereq gems fail to install

### PR DESCRIPTION
**Pull request by indirect**
One of the causes of builds that fail late is gems that didn't install but also didn't raise an error (until the tests that depend on them fail, much much later). This tries to fail those quickly, and maybe means that we can add some retry logic.

### CODE REVIEW

**Comment by indirect** [comment]
@homu r+ (I want to see how the tests do, this still needs code review)

**Comment by indirect** [comment]
@homu r+

File: spec/support/sometimes.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+module Sometimes
+  def run_with_retries(example_to_run, retries)
+    example = RSpec.current_example
```
Comment (segiddins) [inline spec/support/sometimes.rb]: couldn't this just be `example_to_run`?

File: spec/support/sometimes.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    retried_examples = RSpec.world.example_groups.map do |g|
+      g.descendants.map do |d|
+        d.filtered_examples.select do |e|
+          e.metadata[:sometimes] && e.metadata.fetch(:retried, 1) > 1
+        end
+      end
+    end.flatten
+    message.call(retried_examples.empty? ? :green : :yellow, "\n\nRetried examples: #{retried_examples.count}")
+    unless retried_examples.empty?
```
Comment (segiddins) [inline spec/support/sometimes.rb]: no need for this check

File: spec/support/sometimes.rb (hunk 1 above)
Comment (indirect) [inline spec/support/sometimes.rb]: no, they are inexplicably different objects. :'(

File: spec/support/sometimes.rb (hunk 2 above)
Comment (indirect) [inline spec/support/sometimes.rb]: oh yeah, good catch

**Comment by indirect** [comment]
@segiddins r?

**Comment by segiddins** [comment]
@homu r+

**Pull request merged**